### PR TITLE
[SPARK-24224][ML-Examples]Java example code for Power Iteration Clustering  in spark.ml

### PR DESCRIPTION
**Pull request by shahidki31**
## What changes were proposed in this pull request?

Java example code for Power Iteration Clustering  in spark.ml

## How was this patch tested?

Locally tested

### CODE REVIEW

File: examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.examples.ml;
+
+// $example on$
+
```
Comment (hhbyyh) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java]: unnecessary line.

File: examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+import java.util.Arrays;
+import java.util.List;
```
Comment (hhbyyh) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java]: order of import. move java ones to top.

File: examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
```
Comment (hhbyyh) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java]: unnecessary line

File: examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                .getOrCreate();
+
+        // $example on$
+        // Creates data.
+        List<Row> data = Arrays.asList(
+                RowFactory.create(0L, Arrays.asList(1L), Arrays.asList(0.9)),
+                RowFactory.create(1L, Arrays.asList(2L), Arrays.asList(0.9)),
+                RowFactory.create(2L, Arrays.asList(3L), Arrays.asList(0.9)),
+                RowFactory.create(3L, Arrays.asList(4L), Arrays.asList(0.1)),
+                RowFactory.create(4L, Arrays.asList(5L), Arrays.asList(0.9))
```
Comment (hhbyyh) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java]: Maybe we could increase the complexity of the data by a little bit, e.g. put more than one elements in some `neighbors` and `similarities`

File: examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    public static void main(String[] args) {
+        // Create a SparkSession.
+        SparkSession spark = SparkSession
+                .builder()
+                .appName("JavaPowerIterationClustering")
+                .getOrCreate();
+
+        // $example on$
+        // Creates data.
```
Comment (hhbyyh) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java]: for consistency: Either remove the detailed comments or add one for training.

File: examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                .setK(2)
+                .setMaxIter(10);
+
+        Dataset<Row> result = pic.transform(df);
+
+        // printing results
+        System.out.println("Clustering results [id , cluster]");
+        for (Row row : result.select("id","prediction").collectAsList()) {
+            System.out.println("[" + row.get(0) + " , " + row.get(1) + "]");
+        }
```
Comment (hhbyyh) [inline examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java]: Can we just use `show` here?

File: examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java (hunk 1 above)
Comment (shahidki31) [reply examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java]: @hhbyyh Removed the unnecessary line

File: examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java (hunk 2 above)
Comment (shahidki31) [reply examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java]: Rearranged the import

File: examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java (hunk 3 above)
Comment (shahidki31) [reply examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java]: Removed the unnecessary lines

File: examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java (hunk 4 above)
Comment (shahidki31) [reply examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java]: Thanks, I have increased the complexity of the data by putting more elements in the neighbors and similarities

File: examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java (hunk 5 above)
Comment (shahidki31) [reply examples/src/main/java/org/apache/spark/examples/ml/JavaPowerIterationClusteringExample.java]: Done

**Comment by shahidki31** [comment]
Thanks @hhbyyh for the review comments. I have modified the example code based on the review.

**Review by srowen — CHANGES_REQUESTED** [review]
This needs to use 2-space indents.

**Comment by srowen** [comment]
This still doesn't seem to use the right indentation. Have a look at how the other examples are formatted.

**Comment by shahidki31** [comment]
Hi @srowen , Thanks for the comment. I have modified the code.

**Comment by srowen** [comment]
Merged to master

**Comment by shahidki31** [comment]
Thanks @srowen

**Pull request closed without merging**